web/generate-locales-list.js

```diff
@@ -0,0 +1,19 @@
+const fs = require('fs');
+
+const files = fs.readdirSync('./locales')
+  .filter(file => file.endsWith('.json'));
+
+const locales = files.map(file => file.replace('.json', ''));
+
+const localesString = JSON.stringify(locales)
+  .replace(/"/g, '\'')
+  .replace(/,/g, ', ');
+
+fs.writeFileSync('./locales/locales-list.js', `module.exports = ${localesString};\n`);
```

Comment:
it would make more sense to store it in `src/` (it's not a locale file, so it shouldn't live in locales)